### PR TITLE
[New Package] libssp

### PR DESCRIPTION
**Pull request by saltedcoffii**
I'll add more packages as I get them to work, and then remove conflicts from the main package. Hopefully this can solve a lot of our problems with gcc as a core package.

### CODE REVIEW

**Comment by saltedcoffii** [comment]
Actually, @satmandu let me know if the most recent version of gcc has libssp in it. If not, this package won't conflict with libssp and we can add it.

File: packages/libssp.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  @gcc_name = 'libssp'
+  @gcc_global_opts = '--enable-static \
+                      --enable-shared \
+                      --disable-multilib \
+                      --disable-cet \
+                      --with-pic \
+                      --enable-symvers \
+                      --with-gcc-major-version-only \
+                      --with-system-libunwind'
+  @gcc_global_env = "#{CREW_ENV_OPTIONS}"
```
Comment (supechicken) [inline packages/libssp.rb:24]: This can be simplify as `@gcc_global_env = CREW_ENV_OPTIONS`

**Comment by satmandu** [comment]
I have been building GCC 10.3.0 w/o libssp..

**Comment by saltedcoffii** [comment]
Cool! So libssp can be its own package

File: packages/libssp.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+require 'package'
+
+class Libssp < Package
+  description 'Libssp is a part of the GCC toolkit.'
+  homepage 'https://gcc.gnu.org/'
+  version '10.3.0'
+  license 'GPL-3, LGPL-3, libgcc, FDL-1.2'
```
Comment (uberhacker) [inline packages/libssp.rb:7]: How can so many licenses exist for this?  It's confusing.

File: packages/libssp.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                      --disable-cet \
+                      --with-pic \
+                      --enable-symvers \
+                      --with-gcc-major-version-only \
+                      --with-system-libunwind'
+
+  def self.build
+    Dir.mkdir "#{@gcc_name}-builddir"
+    Dir.chdir "#{@gcc_name}-builddir" do
+      system "#{CREW_ENV_OPTIONS} ../#{@gcc_name}/configure #{CREW_OPTIONS} \
```
Comment (uberhacker) [inline packages/libssp.rb:26]: `system "env #{CREW_ENV_OPTIONS}`?

**Comment by satmandu** [comment]
> Cool! So libssp can be its own package

Yes! But if we can get packages to compile without it, we may not have much use of it.

At some point the granularity gets to the point where we have a package per file. Where do we stop?

File: packages/libssp.rb (hunk 2 above)
Comment (saltedcoffii) [reply packages/libssp.rb:7]: These are all the licenses for GCC, I don't know libssp's licenses specifically

File: packages/libssp.rb (hunk 3 above)
Comment (saltedcoffii) [reply packages/libssp.rb:26]: Makes no difference. Do you really want it changed?

**Comment by saltedcoffii** [comment]
Well, libssp does provide some safety stuff for ruby. Apparently the reason it links to libssp in the first place was issues with segfaulting on some platforms.

**Comment by satmandu** [comment]
I still want a way to do this as a single gcc compile and then split the compiled output into multiple packages... 

Having said that, I'm letting this compile overnight for all 3 architectures...

**Comment by satmandu** [comment]
Added binaries

**Comment by saltedcoffii** [comment]
Okay, I think this is ready to merge

**Pull request merged**